File: 26.ts
function removeDuplicates(nums: number[]): number {
	let fast = 1
	let slow = 1

	for (; fast < nums.length; fast++) {
		if (nums[fast] != nums[fast-1]) {
			nums[slow] = nums[fast]
			slow++
		}
	}

	return slow
};

File: 26.go
func removeDuplicates(nums []int) int {
	// 快慢指针法

	n := len(nums)
	if n == 0 {
		return 0
	}

    fast := 1
	slow := 1

	for ; fast < n; fast++ {
		if nums[fast] != nums[fast-1] {
			nums[slow] = nums[fast]
			slow++
		}
	}

	return slow
}